src/mapValues.test.ts

```diff
@@ -160,4 +160,24 @@ describe("typing", () => {
       d: boolean;
     }>();
   });
+
+  test("unions of records", () => {
+    const dataFirst = mapValues(
+      {} as Record<number, string> | Record<string, string>,
```

Comment:
similar to previous comments: 1. pull a `data` variable out, and use `unknown` in places we don't care about a specific type, like the values here.